sql/analyzer/indexes.go

```diff
@@ -904,3 +954,15 @@ func canMergeIndexes(a, b sql.IndexLookup) bool {
 
 	return m.IsMergeable(b)
 }
+
+// convertIsNullForIndexes converts all nested IsNull(col) expressions to Equals(col, nil) expressions, as they are
+// equivalent as far as the index interfaces are concerned.
+func convertIsNullForIndexes(e sql.Expression) (sql.Expression, error) {
```

Comment:
Looks like this can't actually return an error, right? Could simplify the return type